frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/preferences/UserPreferencesPortlet.java

```diff
@@ -44,12 +56,38 @@ private void prepareStandardView(RenderRequest request) {
         final User user = UserCacheHolder.getRefreshedUserFromEmail(UserCacheHolder.getUserFromRequest(request).getEmail());
         SW360Utils.initializeMailNotificationsPreferences(user);
         request.setAttribute(PortalConstants.SW360_USER, user);
-
         request.setAttribute("eventsConfig", SW360Constants.NOTIFIABLE_ROLES_BY_OBJECT_TYPE);
+        request.setAttribute("accessTokenList", CommonUtils.nullToEmptyList(user.getRestApiTokens()));
+    }
+
+    @UsedAsLiferayAction
+    public void createToken(ActionRequest request, ActionResponse response) {
+        User user = UserCacheHolder.getRefreshedUserFromEmail(UserCacheHolder.getUserFromRequest(request).getEmail());
+        UserService.Iface userClient = thriftClients.makeUserClient();
+        RestApiToken restApiToken = getApiTokenInfoFromRequest(request);
+        if (isDuplicateTokenName(user, restApiToken.getName())) {
+            log.error("Token name [" + restApiToken.getToken() + "] already exists for user " + user.getEmail());
+            setSW360SessionError(request, ErrorMessages.REST_API_TOKEN_NAME_DUPLICATE);
+            return;
+        }
+
+        String token = RandomStringUtils.random(16, true, true);
+        restApiToken.setToken(BCrypt.hashpw(token, BCrypt.gensalt()));
+        restApiToken.setCreatedOn(SW360Utils.getCreatedOnTime());
+        user.addToRestApiTokens(restApiToken);
+
+        try {
+            userClient.updateUser(user);
+            request.setAttribute("accessToken", token);
+            request.setAttribute("accessTokenList", CommonUtils.nullToEmptyList(user.getRestApiTokens()));
```

Comment:
CommonUtils.nullToEmptyList not necessary here because `user.getRestApiTokens()` is always > 1 in this step.